fe/fe-core/src/main/java/com/starrocks/mysql/MysqlProto.java

```diff
@@ -139,6 +139,13 @@ public static NegotiateResult negotiate(ConnectContext context) throws IOExcepti
             return new NegotiateResult(null, NegotiateState.READ_FIRST_AUTH_PKG_FAILED);
         }
 
+        if (Config.require_secure_transport && !authPacket.isSSLConnRequest()) {
```

Comment:
How about adding it to 
if (authPacket.isSSLConnRequest()) {
} else if (Config.require_secure_transport) {
// add code here.
}

And better to allow localhost to login without ssl encryption.

Reply:
Thanks, @gengjun-git, for checking. That makes sense.

Reply:
Good job!!
And there is another suggestion you seem to have missed.
Better to allow localhost to login without ssl encryption. 
You can get client ip using `context.getRemoteIP()`

Reply:
@gengjun-git, by default, the `Config.ssl_force_secure_transport` is set to `false`. This means that unless SSL is explicitly enabled or `Config.ssl_force_secure_transport` is set to `true` locally, logging in from localhost is permitted without SSL encryption.

I may be misunderstanding your comments. Could you please clarify the requested change regarding localhost login?

Reply:
@rstyp I mean, even if Config.ssl_force_secure_transport is set to true, and localhost wants to login without ssl, we should allow the login. 
This is for operation and maintenance. If there is a problem with the SSL certificate configuration, we can login from   localhost to make some changes.